internal/lbapi/client.go

```diff
@@ -0,0 +1,114 @@
+package lbapi
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/hashicorp/go-retryablehttp"
+)
+
+const (
+	apiVersion = "v1"
+)
+
+// HTTPClient interface
+type HTTPClient interface {
+	Do(req *retryablehttp.Request) (*http.Response, error)
+}
+
+type Client struct {
+	client  HTTPClient
+	baseURL string
+}
+
+func NewClient(url string, opts ...func(*Client)) *Client {
```

Comment:
you're taking functional arguments here but you aren't using them (and none exist) - probably you should at least complete the pattern in the function:

```golang
	for _, opt := range opts {
		opt(&client)
	}
```

 or change `opts` to `_` until you implement it. I assume you want to either be able to pass the http client or the retry params eventually.

Reply:
this one was a bit gnarly to be able to mock the HTTPClient client for testing. I originally had some client override functions implemented, but then removed them.